lib/topic_query/private_message_lists.rb

```diff
@@ -0,0 +1,265 @@
+# frozen_string_literal: true
+
+class TopicQuery
+  module PrivateMessageLists
+    def list_private_messages_all(user)
+      list = private_messages_for(user, :all)
+      list = filter_archived(list, user, archived: false)
+      create_list(:private_messages, {}, list)
+    end
+
+    def list_private_messages_all_sent(user)
+      list = private_messages_for(user, :all)
+
+      list = list.where(<<~SQL, user.id)
+      EXISTS (
+        SELECT 1 FROM posts
+        WHERE posts.topic_id = topics.id AND posts.user_id = ?
+      )
+      SQL
+
+      list = filter_archived(list, user, archived: false)
+      create_list(:private_messages, {}, list)
+    end
+
+    def list_private_messages_all_archive(user)
+      list = private_messages_for(user, :all)
+      list = filter_archived(list, user, archived: true)
+      create_list(:private_messages, {}, list)
+    end
+
+    def list_private_messages_all_new(user)
+      list_private_messages_new(user, :all)
+    end
+
+    def list_private_messages_all_unread(user)
+      list_private_messages_unread(user, :all)
+    end
+
+    def list_private_messages(user)
+      list = private_messages_for(user, :user)
+      list = not_archived(list, user)
+      create_list(:private_messages, {}, list)
+    end
+
+    def list_private_messages_archive(user)
+      list = private_messages_for(user, :user)
+      list = list.joins(:user_archived_messages).where('user_archived_messages.user_id = ?', user.id)
+      create_list(:private_messages, {}, list)
+    end
+
+    def list_private_messages_sent(user)
+      list = private_messages_for(user, :user)
+
+      list = list.where(<<~SQL, user.id)
+      EXISTS (
+        SELECT 1 FROM posts
+        WHERE posts.topic_id = topics.id AND posts.user_id = ?
+      )
+      SQL
+
+      list = not_archived(list, user)
+      create_list(:private_messages, {}, list)
+    end
+
+    def list_private_messages_new(user, type = :user)
+      list = TopicQuery.new_filter(
+        private_messages_for(user, type),
+        treat_as_new_topic_start_date: user.user_option.treat_as_new_topic_start_date
+      )
+
+      list = remove_muted_tags(list, user)
+
+      create_list(:private_messages, {}, list)
+    end
+
+    def list_private_messages_unread(user, type = :user)
+      list = TopicQuery.unread_filter(
+        private_messages_for(user, type),
+        staff: user.staff?
+      )
+
+      first_unread_pm_at = UserStat
+        .where(user_id: user.id)
+        .pluck_first(:first_unread_pm_at)
+
+      if first_unread_pm_at
+        list = list.where("topics.updated_at >= ?", first_unread_pm_at)
+      end
+
+      create_list(:private_messages, {}, list)
+    end
+
+    def list_private_messages_group(user)
+      list = private_messages_for(user, :group)
+
+      list = list.joins(<<~SQL)
+      LEFT JOIN group_archived_messages gm
+      ON gm.topic_id = topics.id AND gm.group_id = #{group&.id&.to_i}
```

Comment:
Is the intention here to default to group 0 "everyone" if `group` is nil? It seems peculiar to me that we'd have private messages to "Everyone" and maybe this is not what we want? It actually seems quite odd to me to call `list_private_messages_group` if we have no `group`. Maybe it should raise an error instead?

Reply:
(same comment applies in the other `_group` methods below.)

Reply:
Good catch, I was copying and pasting code when refactoring the methods. If the group is missing which we don't expect since the controller would have handled it, `TopicQuery` should just raise an error. I've updated the code to remove any expectations that `group` might be `nil`.